sbi/inference/snpe/snpe_base.py

```diff
@@ -376,6 +290,103 @@ def _train(
         self._summary["epochs"].append(epoch)
         self._summary["best_validation_log_probs"].append(self._best_val_log_prob)
 
+        # Update tensorboard and summary dict.
+        self._summarize(
+            round_=self._round, x_o=None, theta_bank=theta, x_bank=x,
+        )
+
+        # Update description for progress bar.
+        if show_train_summary:
+            print(self._describe_round(self._round, self._summary))
+
+        return deepcopy(self._neural_net)
+
+    def build_posterior(
+        self,
+        density_estimator: Optional[TorchModule] = None,
```

Comment:
same comment like for SNLE. I find it confusing to be able to pass a density estimator to replace the trained one. same for the posterior. I might be missing something, but I would vote for moving this a separate, potentially unbound method.

Reply:
This was resolved (?) but in sentiment I agree with Jan.